lib/t/matches.ex

```diff
@@ -601,24 +736,75 @@ defmodule T.Matches do
   end
 
   def match_check() do
-    MatchEvents
+    MatchEvent
+    |> order_by(desc: :timestamp)
     |> distinct([m], m.match_id)
     |> join(:inner, [m], ma in Match, on: ma.id == m.match_id)
     |> where([m], m.timestamp < fragment("now() - INTERVAL '48 hours'"))
-    |> select([m, ma], {m.match_id, ma.user_id_1})
+    |> select([m, ma], {m.match_id, ma.user_id_1, ma.user_id_2})
     |> T.Repo.all()
-    |> Enum.map(fn {match_id, user_id} ->
-      T.Matches.unmatch_match(user_id, match_id)
+    |> Enum.map(fn {_match_id, user_id_1, user_id_2} ->
+      T.Matches.expire_match(user_id_1, user_id_2)
     end)
   end
 
   def match_timeslot_new_event(match_id, event) do
     if match_id != [] and match_id != nil and match_id != "" do
-      Repo.insert(%MatchEvents{
+      Repo.insert(%MatchEvent{
         timestamp: DateTime.truncate(DateTime.utc_now(), :second),
         match_id: match_id,
         event: event
       })
     end
   end
+
+  def mark_expired_match_seen(match_id, by_user_id) do
+    ExpiredMatch
+    |> where(match_id: ^match_id, user_id: ^by_user_id)
+    |> Repo.delete_all()
+  end
+
+  def expiration_date(match_id) do
+    successfull_calls =
+      MatchEvent
+      |> where(match_id: ^match_id)
+      |> where(event: "call start")
+      |> select([e], e.match_id)
+
+    MatchEvent
+    |> where([e], e.match_id not in subquery(successfull_calls))
+    |> where(match_id: ^match_id)
+    |> distinct([e], e.match_id)
+    |> order_by(desc: :timestamp)
+    |> select([e], e.timestamp)
+    |> Repo.one()
+    # TODO TO ENV VARS
+    |> DateTime.add(2 * 24 * 60 * 60)
+  end
+
+  def expiration_date(user_id_1, user_id_2) do
+    [uid1, uid2] = Enum.sort([user_id_1, user_id_2])
+    match_id = T.Matches.get_match_id_for_users!(uid1, uid2)
+
+    successfull_calls =
+      MatchEvent
+      |> where(match_id: ^match_id)
+      |> where(event: "call start")
+      |> select([e], e.match_id)
+
+    MatchEvent
+    |> where([e], e.match_id not in subquery(successfull_calls))
+    |> where(match_id: ^match_id)
+    |> distinct([e], e.match_id)
+    |> order_by(desc: :timestamp)
+    |> select([e], e.timestamp)
+    |> Repo.one()
+    # TODO TO ENV VARS
+    |> DateTime.add(2 * 24 * 60 * 60)
+  end
+
+  # defp schedule_match_about_to_expire(match_id) do
```

Comment:
(commented it out to silence warning about function being unused)

@orlovvlad95 could you please create a pruner system (similar to the one which expires matches) which will send push notifications using this function 2 hours before match is about to expire?